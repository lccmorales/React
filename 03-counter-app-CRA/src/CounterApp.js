import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
    const [ counter, setCounter ] = useState( value );

    //handleAdd
    const handleAdd = () => {
        setCounter ( counter + 1 );
        // setCounter((c) => c + 1);
    }

    //handleReset
    const handleReset = () => setCounter ( value );
    

    //handleSub
    const handleSub = () => setCounter ( counter - 1 );
    

    return (
      <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button onClick = { handleAdd } >+1</button>
        <button onClick = { handleReset } >Reset</button>
        <button onClick = { handleSub } >-1</button>
      </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;