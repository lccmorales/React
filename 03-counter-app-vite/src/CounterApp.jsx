import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState( value );

    //handleReset
    const handleReset = () => setCounter ( value );
    
    //handleSub
    const handleSub = () => setCounter ( counter - 1 );
    
    //handleAdd
    const handleAdd = () => {
        setCounter ( counter + 1 );
        // setCounter((c) => c + 1);
    }

    return (
      <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <button onClick = { handleSub } >-1</button>
        <button onClick = { handleReset } aria-label="btn-reset" >Reset</button>
        <button onClick = { handleAdd } >+1</button>
      </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 25
}