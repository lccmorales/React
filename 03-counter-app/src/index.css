html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, monospace;
    font-size: 1.0rem;
    /* padding: 70px; */
}

button {
    font-size: 1.0rem;
    margin-right: 10px;
    padding: 5px;
}