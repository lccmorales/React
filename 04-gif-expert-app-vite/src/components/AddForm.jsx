import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddForm = ( { setCategories, setLimitNumGifs } ) => {
    const [inputValueSearch, setInputValueSearch] = useState('');
    const [inputValueLimit, setInputValueLimit] = useState('');

    const handleInputSearchChange = (e) => {
        setInputValueSearch(e.target.value);
        console.log('handleInputSearchChange - llamado');
    }

    const handleInputLimitChange = (e) => {
        setInputValueLimit(e.target.value);
        console.log('handleInputLimitChange - llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit - llamado', inputValueSearch);
        if (inputValueSearch.trim().length > 3 && parseInt(inputValueLimit.trim()) > 0) {
            setCategories( cats => [ inputValueSearch, ...cats,] );
            setInputValueSearch('');
            setLimitNumGifs( parseInt(inputValueLimit) );
            setInputValueLimit('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValueSearch }</p>
            <p>{ inputValueLimit }</p>
            <h3>Busqueda</h3>
            <input
                className='search'
                type="text"
                value={ inputValueSearch }
                onChange={ handleInputSearchChange }
            />

            <h3>Cantidad de Gif's a mostrar</h3>
            <input
                type="number"
                value={ inputValueLimit }
                onChange={ handleInputLimitChange }
            />

            <button type="submit">Buscar</button>
        </form>
    )
}

AddForm.propTypes = {
    setCategories: PropTypes.func.isRequired,
    setLimitNumGifs: PropTypes.func.isRequired
};