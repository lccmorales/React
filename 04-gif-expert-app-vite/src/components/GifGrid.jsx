import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category, limitNumGifs } ) => {
    const { images, loading } = useFetchGifs( category, limitNumGifs );

    return (
        <>
        <h3 className='animate__animated animate__flash' aria-label='titulo'>{ category }</h3>

        { loading && <p className='animate__animated animate__flash'>Cargando...</p> }
        
        <div className='card-grid'>
            {
                images.map( img => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limitNumGifs: PropTypes.number.isRequired
}