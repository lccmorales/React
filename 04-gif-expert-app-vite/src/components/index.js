export * from './AddForm';
export * from './GifGrid';
export * from './GifGridItem';