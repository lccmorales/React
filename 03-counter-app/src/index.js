import React from 'react';
// import ReactDOM from 'react-dom';
import ReactDomClient from 'react-dom/client';
// import PrimeraApp from './PrimeraApp';
import './index.css';
import CounterApp from './CounterApp';

const divRoot = document.querySelector('#root');
// ReactDOM.render(<PrimeraApp />, divRoot);
const root = ReactDomClient.createRoot(divRoot);
// root.render (<PrimeraApp saludo="Hola Mundo" />);
root.render (<CounterApp value={ 10 } />);