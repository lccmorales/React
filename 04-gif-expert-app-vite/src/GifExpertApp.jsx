import { useState } from "react";
import { AddForm, GifGrid } from './components';

export const GifExpertApp = ( { defaultCategories = []} ) => {
    const [ categories, setCategories ] = useState( defaultCategories );
    const [ limitNumGifs, setLimitNumGifs ] = useState(2);
    
    const onAddForm = ( newCategory, newLimitNumGif ) => {
        console.log(newCategory, ' - ', newLimitNumGif);
        if ( categories.includes(newCategory) )  return;
        setCategories([ newCategory, ...categories ]);
        setLimitNumGifs( parseInt(newLimitNumGif) );
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            {/* <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs }/> */}
            <AddForm onNewForm = { (cat, limit) => onAddForm(cat, limit) }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category } limitNumGifs={ limitNumGifs }/>
                    ))
                }
            </ol>
        </>
    );
}