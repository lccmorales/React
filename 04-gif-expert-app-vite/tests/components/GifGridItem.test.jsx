import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components';

describe('Pruebas en <GifGridItem />', () => {
    const title = 'Saitama';
    const url = 'http://one-punch.com/saitama.jpg';
    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifGridItem title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('debe de mostrar el titulo del componente', () => {
        render(<GifGridItem title={ title } url={ url } />);
        expect( screen.getByText( title ) ).toBeTruthy();
    });
})
