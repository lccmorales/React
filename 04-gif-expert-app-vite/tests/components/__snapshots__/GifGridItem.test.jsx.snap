// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__swing"
  >
    <img
      alt="Saitama"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
